src/consensus/aft/raft.h

```diff
@@ -2640,14 +2671,25 @@ namespace aft
       }
     }
 
-    void become_leader()
+    void become_leader(bool force_become_leader = false)
     {
       if (is_retired())
       {
         return;
       }
 
-      election_index = last_committable_index();
+      // When we force to become the primary we are going are around the
+      // consensus protocol. Therefore in BFT we need to bypass the information in
+      // the progress tracker when working out how far we rollback.
```

Comment:
I get that, but I would have thought that we only do this once at startup, when the progress tracker has a well known initial state. Am I missing something?

Reply:
no this is exactly what we are doing

Reply:
Ok, this is so we don't roll back the genesis/recovery tx, I see.